### PR TITLE
feat(vite): add createNodesV2 function

### PR DESCRIPTION
**Pull request by Coly010**
- chore(vite): move single file processing of plugin to own function
- feat(vite): add CreateNodesV2 function

## Current Behavior

There is no implementation for createNodesV2

## Expected Behavior

There should be an implementation for createNodesV2

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/vite/src/plugins/plugin.ts [hunk 1]
```diff
@@ -23,76 +28,123 @@ export interface VitePluginOptions {
   previewTargetName?: string;
   serveStaticTargetName?: string;
 }
+type ViteTargets = Pick<ProjectConfiguration, 'targets' | 'metadata'>;
 
 const cachePath = join(workspaceDataDirectory, 'vite.hash');
 const targetsCache = readTargetsCache();
```
Comment (FrozenPandaz) [inline packages/vite/src/plugins/plugin.ts:34]: Don't read this in the moduleScope, there's no point.

File: packages/vite/src/plugins/plugin.ts [hunk 2]
```diff
@@ -23,76 +28,123 @@ export interface VitePluginOptions {
...
+    const oldCache = readTargetsCache();
+    writeJsonFile(cachePath, {
+      ...oldCache,
+      ...targetsCache,
+    });
+  }
 }
 
 export const createDependencies: CreateDependencies = () => {
   writeTargetsToCache();
```
Comment (FrozenPandaz) [inline packages/vite/src/plugins/plugin.ts:58]: This is not needed anymore.

File: packages/vite/src/plugins/plugin.ts [hunk 3]
```diff
@@ -23,76 +28,123 @@ export interface VitePluginOptions {
...
-    if (
-      !siblingFiles.includes('package.json') &&
-      !siblingFiles.includes('project.json')
-    ) {
-      return {};
-    }
+    logger.warn(
+      '`createNodes` is deprecated. Update your plugin to utilize createNodesV2 instead. In Nx 20, this will change to the createNodesV2 API.'
+    );
+    return createNodesInternal(configFilePath, options, context, targetsCache);
```
Comment (FrozenPandaz) [inline packages/vite/src/plugins/plugin.ts:90]: It's okay to use an `{}` here for `targetsCache`.

This function shouldn't really get called unless it's a rare case that someone wrapped our plugin `createNodes` function.

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

**Review by jaysoo — APPROVED** [review]

File: packages/vite/src/plugins/plugin.ts [hunk 4]
```diff
@@ -23,76 +28,109 @@ export interface VitePluginOptions {
...
-    ...targetsCache,
-  });
+function writeTargetsToCache(cachePath, results?: Record<string, ViteTargets>) {
+  writeJsonFile(cachePath, results);
 }
 
 export const createDependencies: CreateDependencies = () => {
-  writeTargetsToCache();
+  logger.warn(
+    "The 'createDependencies' function is now a no-op. It will be replaced via the 'CreateNodesV2' API."
```
Comment (FrozenPandaz) [inline packages/vite/src/plugins/plugin.ts:43]:
```suggestion
    "The 'createDependencies' function is now a no-op. This functionality is included in 'createNodesV2'."
```

This has to be moved to a jsDoc.

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**